pkg/settlement/swap/chequebook/cheque.go

```diff
@@ -0,0 +1,94 @@
+package chequebook
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/crypto/eip712"
+)
+
+// Cheque represents a cheque for a SimpleSwap chequebook
+type Cheque struct {
+	Chequebook       common.Address
+	Beneficiary      common.Address
+	CumulativePayout *big.Int
+}
+
+// SignedCheque represents a cheque together with its signature
+type SignedCheque struct {
```

Comment:
i wonder if we need both types. a `Cheque` could just have the signature field, and have a helper method `.Signed()` that returns a bool to indicate whether it is signer or not

Reply:
Not sure I see the advantage this would bring. Having these types separate makes it explicit at a type level where we expect a signature to be present and where not.